[W7.7][W14-B1] iContacts

tshradheya opened the pull request:
V1.5rc Released.

Ready for Review

Overview of V1.5rc

Features added: 

* Import command
* Enhanced UI
* Fixed Bugs
* Details command to improve UI

Documentation 

* UG updated as necessary
* Features implementation described in DG
* Portfolio added
* Code collated

Used Github to manage our [milestones for V1.5rc](https://github.com/CS2103AUG2017-W14-B1/main/milestone/4?closed=1) [ Issues, Milestones]

The [release made with V1.5rc Tag](https://github.com/CS2103AUG2017-W14-B1/main/releases/tag/V1.5rc)

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -714,42 +714,127 @@ We now have everything set up... but we still can't modify the remarks. Let's fi
+
+|`* * *` |user |sort the list by name alphabetically |view contacts in an alphabetical order
+
+|`* * *` |user with limited details of a contact |add user with just the name and phone number |store contact information with limited detail
```
may be reword this? you don't specify what limited details you have, but want to add name and phone number

tshradheya replied:
made the changes. Limited details include name and phone number

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -763,20 +848,197 @@ Use case ends.
+*MSS*
+
+1.  User selects contacts to be exported based on current list
+2.  User requests for creating file containing selected contacts
```
use cases should capture the user-system interaction. in light of this, there should be something between 1 and 2 that the system does

tshradheya replied:
Agreed. Taken care of

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -763,20 +848,197 @@ Use case ends.
+* 4a. iContacts finds duplicate contacts
++
+[none]
+** 4a1. iContacts shows an error message and doesn't merge the duplicate ones
```
does iContacts stop merging completely? or only duplicates are not merged? can be specific here

tshradheya replied:
Only duplicate ones are not merged. Made the necessary change

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -763,20 +848,197 @@ Use case ends.
+*MSS*
+
+1.  User specifies the keywords for different link:#attributes[attributes] and requests the list
+2.  iContacts finds list of all people satisfying all attributes
```
may be combine 2 and 3?

tshradheya replied:
Agreed. Done

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -763,20 +848,197 @@ Use case ends.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
-
-{More to be added}
+.  A new user to a link:#CLI[CLI] environment and/or address book application should be able to learn how to install and use the application within 20 minutes with the help of the User Guide.
```
our constraints say, application should require no installation.

tshradheya replied:
Noted. Changes made

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -763,20 +848,197 @@ Use case ends.
-
-{More to be added}
+.  A new user to a link:#CLI[CLI] environment and/or address book application should be able to learn how to install and use the application within 20 minutes with the help of the User Guide.
+.  Should be responsive and commands are executed within 2 seconds under normal load (1000 persons or less) and typical usage.
```
what is typical usage? can you make it objective?

tshradheya replied:
The implication of typical usage is quite ambiguous so decided to omit the part

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -763,20 +848,197 @@ Use case ends.
+.  A new user to a link:#CLI[CLI] environment and/or address book application should be able to learn how to install and use the application within 20 minutes with the help of the User Guide.
+.  Should be responsive and commands are executed within 2 seconds under normal load (1000 persons or less) and typical usage.
+.  Essential services (adding, deleting contacts etc) should still work perfectly without Internet access.
+.  Layout of the application should be clear, appealing and attractive.
```
define `clear`, `appealing` and `attractive`

tshradheya replied:
Done. Defined in glossary after making NFR more objective

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -763,20 +848,197 @@ Use case ends.
+.  Should be responsive and commands are executed within 2 seconds under normal load (1000 persons or less) and typical usage.
+.  Essential services (adding, deleting contacts etc) should still work perfectly without Internet access.
+.  Layout of the application should be clear, appealing and attractive.
+.  Should be able to handle any invalid input and recover from it.
```
isn't this more of the functionality?

tshradheya replied:
Agreed. NFR deleted

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -763,20 +848,197 @@ Use case ends.
+.  Essential services (adding, deleting contacts etc) should still work perfectly without Internet access.
+.  Layout of the application should be clear, appealing and attractive.
+.  Should be able to handle any invalid input and recover from it.
+.  Should not consume too much processing time/power, less than 10% of the CPU of modern computers/laptops.
```
how much is too much?

tshradheya replied:
made it more objective

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -763,20 +848,197 @@ Use case ends.
+.  Should be backwards compatible with storage files from prior versions.
+.  Should be compatible with storage files from the same application but from different computers.
+.  Should minimize dependencies on third-party software.
+.  Should consistently come up with at least one new feature every month.
```
this is more of the process, not really NFR

tshradheya replied:
removed

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -763,20 +848,197 @@ Use case ends.
+.  Should be compatible with storage files from the same application but from different computers.
+.  Should minimize dependencies on third-party software.
+.  Should consistently come up with at least one new feature every month.
+.  Should require some form of authentication before allowing usage.
```
i guess the ones below are all related to process or design. not really NFR

tshradheya replied:
Agreed. Removed the ones which aren't NFR's

okkhoy commented:
@tshradheya @chuaweiwen @justinpoh @edwinghy 
I don't see your code enhancement PRs merged to your master. Please do so!

some comments added. please read and ack; please retain the PR without closing.

okkhoy commented on docs/UserGuide.adoc:
```diff
@@ -91,6 +98,85 @@ Examples:
+image::directon.png[width="790"]
+
+
+=== Email directly to a group of people having a particular tag: `email` [Coming in v2.0]
```
:+1: marking it as in V2.0

okkhoy commented on docs/DeveloperGuide.adoc:
```diff
@@ -714,42 +714,127 @@ We now have everything set up... but we still can't modify the remarks. Let's fi
+
+This application is targeted towards University students, aiming to meet their needs throughout their university life, and hopefully beyond. University students meet many people in the university, such as project group members, tutors, lecturers, friends and other connections. Nowadays it is important for students to build connections with other people, as connections may also help students for t ...
+
+*  User Profile
```
this is nicely written!  :100:

okkhoy commented:
@tshradheya @justinpoh @chuaweiwen @edwinghy 
Guys, good job on updating the documents appropriately! Well done and keep up the good work.

Please take a look at the few new comments added.

okkhoy commented:
- Please use code snippets sparingly. They are difficult to maintain. Every time your code changes, you will have to update in the documentation.
- Use UML diagrams to show your design or component interaction etc., They are more stable. At present, I don't see any UML diagrams in your implementation section. *(so far)
- This kinda arrow head code is an indicator of bad SLAP. Please check if you can improve it
![image](https://user-images.githubusercontent.com/1023494/31875917-c3badd0a-b802-11e7-8522-c2c9a53a6a6d.png)
- ... String range using the regex ","  ... here `,` is called separator.
- BTW., your code should be self explanatory for such simple things as below:
![image](https://user-images.githubusercontent.com/1023494/31875986-03e111c4-b803-11e7-9c60-f228ab67b470.png)
- Can reduce the size of the image of the wireframe, it looks pixellated now.
- I see good use of sequence diagrams for Toggle mechanism explanation. Good!!

okkhoy commented:
@tshradheya @chuaweiwen @justinpoh @edwinghy  for your action!

tshradheya commented:
V1.2 Review , Team Response: 

* Made changes with respect to the comments in this [PR](https://github.com/CS2103AUG2017-W14-B1/main/pull/145#partial-pull-merging)

1. Removed excessive code snippets.
2. Working on adding more UML diagrams.
3. Corrected the 'code self-explanatory' part.
4. Reduced size of image.
